### ISSUE TITLE
swiftmailer uses starttls with strict certificate checking

### ISSUE THREAD

**Issue by nickhilliard** [inex/IXP-Manager#752]
If STARTTLS fails, it throws up something like the following error:

```
Server Error :: stream_socket_enable_crypto(): Peer certificate CN=`blah.example.com' did not match expected CN=`localhost'
```

There's no clear indication here that the root cause is starttls, or that the documented workaround is to use `MAIL_ENCRYPTION=false` to prevent the error.

Can this exception be caught, parsed and replaced with a UI warning message that SMTP TLS has failed and to check the mail config in .env with a reference to https://docs.ixpmanager.org/usage/email/ ?

**Comment by barryo** [comment]
Relevant - https://laracasts.com/discuss/channels/laravel/stream-socket-enable-crypto-error-laravel?page=1&replyId=349567

UI not a suitable option here as emails can be async and via artisan. 

Catching the exception and providing help in the [test tool](https://docs.ixpmanager.org/usage/email/#testing-smtp) plus docs should cover it.

**Comment by listerr** [comment]
Not sure how relevant this is, but seems Swift Mailer will stop being maintained at the end of November 2021:

https://swiftmailer.symfony.com/docs/introduction.html

But I guess Laravel will switch in due course.

I had a little trouble with this when testing out a new install for 6.2.0, for example when enabling

**Comment by barryo** [comment]
> Not sure how relevant this is, but seems Swift Mailer will stop being maintained at the end of November 2021:
>
> https://swiftmailer.symfony.com/docs/introduction.html
> 
> But I guess Laravel will switch in due course.

Already sorted in the next version: https://github.com/laravel/framework/pull/38481

**Comment by barryo** [comment]
Thanks on the note re `.env.example` and `config/mail.php` being out of sync @listerr. Now fixed in ed8ee84063da69d97aaa16bbb9eb786e9ee8da2f. 

I have also updated the docs: https://docs.ixpmanager.org/usage/email/ *(note there is some information there on testing)*

If you read the docs, you'll note that I'm very much trying to push people down the route of using a local (or internal) SMTP daemon:

> We would generally expect an IXP to have an internal SMTP relay server within the management network to handle the sending of email from monitoring systems, cron processes, IXP Manager, etc.

also:

> MAIL_MAILER="smtp" - this is the mail transport to use. The other available options are outside the scope of this documentation.

As well as trying to reduce the support burden, I actually think this is the better production set-up (and allows for, e.g., a pair of highly resilient mail servers using keepalived).

Hopefully the changed defaults (i.e. localhost smtp without tls) solve most issues.

**Comment by barryo** [comment]
Closing now as:

- [x] Docs include email testing tool - https://docs.ixpmanager.org/usage/email/#testing-smtp
- [x] Verified testing tool provides correct exception: *Error: stream_socket_enable_crypto(): Peer certificate CN=`mail.ibn.ie' did not match expected CN=`46.182.8.12'*
- [x] The ability to disable strict TLS checking is not part of IXP Manager / Laravel (and shouldn't be imho - may as well send plaintext)
- [x] Per @nickhilliard - `.env.example` updated to ref docs and defaults to localhost and no tls
- [x] `.env.example` updated per @listerr's findings also
- [x] Note added to draft release notes for next release on changes
